connectivity/suite.go

```diff
@@ -240,6 +275,124 @@ func Run(ctx context.Context, ct *check.ConnectivityTest) error {
 			return check.ResultDrop, check.ResultNone
 		})
 
+	// Test L7 HTTP introspection using an ingress policy on echo pods.
+	ct.NewTest("echo-ingress-l7-named-port").
+		WithFeatureRequirements(check.RequireFeatureEnabled(check.FeatureL7Proxy)).
+		WithPolicy(echoIngressL7HTTPNamedPortPolicyYAML). // L7 allow policy with HTTP introspection (named port)
+		WithScenarios(
+			tests.PodToPodWithEndpoints(),
+		).
+		WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+			if a.Source().HasLabel("other", "client") { // Only client2 is allowed to make HTTP calls.
+				// Trying to access private endpoint without "secret" header set
+				// should lead to a drop.
+				if a.Destination().Path() == "/private" && !a.Destination().HasLabel("X-Very-Secret-Token", "42") {
+					return check.ResultDropCurlHTTPError, check.ResultNone
+				}
+				egress = check.ResultOK
+				// Expect all curls from client2 to be proxied and to be GET calls.
+				egress.HTTP = check.HTTP{
+					Method: "GET",
+				}
+				return egress, check.ResultNone
+			}
+			return check.ResultDrop, check.ResultNone
+		})
+
+	// Tests with deny policy
+	if ct.DenyPolicy() {
+		// This policy denies ingress into client from client2.
+		ct.NewTest("client-ingress-deny").WithPolicy(clientIngressFromClient2DenyPolicyYAML).
+			WithScenarios(
+				tests.ClientToClient(),
+			).
+			WithExpectations(func(a *check.Action) (egress, ingress check.Result) {
+				if a.Destination().HasLabel("kind", "client") {
+					return check.ResultOK, check.ResultDrop
+				}
+				return check.ResultOK, check.ResultOK
+			})
+
+		// This policy allows ingress to echo only from client with a label 'other:client'
+		ct.NewTest("echo-ingress-deny").WithPolicy(echoIngressFromOtherClientDenyPolicyYAML).
+			WithScenarios(
```

Comment:
Maybe drop this comment about allowing 1.1.1.1 since it conflicts with the comments above and also the policy?

Reply:
Sorry, I'm trying a new code review tool... this is referring to line 389 below.